extensions/apps/settings/src/components/pages/notifications-preferences-option.tsx

```diff
@@ -1,22 +1,89 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { tw } from '@twind/core';
 import Text from '@akashaorg/design-system-core/lib/components/Text';
-import PageLayout from './base-layout';
-import { useAkashaStore, useRootComponentProps } from '@akashaorg/ui-awf-hooks';
+import {
+  useAkashaStore,
+  useNotifications,
+  useRootComponentProps,
+  useTheme,
+} from '@akashaorg/ui-awf-hooks';
 import Stack from '@akashaorg/design-system-core/lib/components/Stack';
 import ErrorLoader from '@akashaorg/design-system-core/lib/components/ErrorLoader';
 import Button from '@akashaorg/design-system-core/lib/components/Button';
 import appRoutes, { PREFERENCES } from '../../routes';
+import Card from '@akashaorg/design-system-core/lib/components/Card';
+import {
+  LockLight,
+  LockDark,
+  InfoLight,
+  InfoDark,
+} from '@akashaorg/design-system-core/lib/components/Icon/akasha-icons';
+import { NotificationEvents, NotificationTypes } from '@akashaorg/typings/lib/ui';
+import Checkbox from '@akashaorg/design-system-core/lib/components/Checkbox';
+import getSDK from '@akashaorg/core-sdk';
+import { UserSettingType } from '@akashaorg/core-sdk/src/common/notification/notification-schemas';
+import { UserSetting } from '@pushprotocol/restapi/src/lib';
+
+const preferencesObjectFactory = (val: boolean): UserSettingType[] => {
+  // Order of items in array determines the setting category in payload (PushOrg api requirement)
+  return [
+    {
+      index: 1,
+      appName: 'Antenna App',
```

Comment:
I suggest to use enum instead of hardcoding the app name then you can create a utility function like `const findAppIndex = (appName:<enum type>) => DEFAULT_PREFERENCES.findIndex(e => e.appName === appName);`

I also suggest to extract the object factory into a separate file along with the above utility function.

Reply:
done 👍